### DISCUSSION TITLE
How to enable full screen on startup?

### DISCUSSION THREAD

**Discussion by theowenyoung** [wezterm/wezterm#2506] [Q&A]
Hi, I'm from alacritty, they have an option:

```yaml
window:
  startup_mode: Fullscreen
```

How can I do this in wezterm?

**Comment by theowenyoung** [comment]
I soved by search, with the following config: 

```lua
local wezterm = require 'wezterm'
local mux = wezterm.mux

wezterm.on('gui-startup', function(window)
  local tab, pane, window = mux.spawn_window(cmd or {})
  local gui_window = window:gui_window();
  gui_window:maximize()
end)
return {
  native_macos_fullscreen_mode = true
}
```

but it's maximized, not fullscreen.

**Reply by wez** [reply]
I just pushed some commits to `main` that will allow you to do this:

```lua
local wezterm = require 'wezterm'
local mux = wezterm.mux

wezterm.on('gui-startup', function(window)
  local tab, pane, window = mux.spawn_window(cmd or {})
  local gui_window = window:gui_window();
  gui_window:perform_action(wezterm.action.ToggleFullScreen, pane)
end)

return {
  native_macos_fullscreen_mode = true
}
```

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Reply by theowenyoung** [reply]
Thanks for the quick response! I  have tried it, and it works fine! Thank you!

**Reply by gegoune** [reply]
Does not work when no multiplexing is active, should it?

**Reply by zw963** [reply]
Hi, just curious, why not add a option to `wezterm start` for start with full screen?

I saw many terminal have this options, e.g.  `konsole --fullscreen`

Or, is there a config option for this? above code is work! but, add those many code for this useful feature is consider can improve?

**Reply by hareki** [reply]
For those who copied the snippet above, and wondered for hours (definitely not me) why `wezterm start [program]` didn't run your specified program,  change `function(window)` to `function(cmd)` at the line `wezterm.on('gui-startup', function(window)`

**Comment by dougg0k** [comment]
```
wezterm.on("gui-startup", function(cmd)
	if mux then
  	local tab, pane, window = mux.spawn_window(cmd or {})
  	window:gui_window():maximize()
	end
end)
```

This cause the first window to open maximized, but if I try to open a second, it start small. How make all window(s) start maximized?

---

Edit:

`--always-new-process` seems to help with a second window.

**Reply by tmccombs** [reply]
I'm running into the same problem.  Is there an equivalent event that is triggered whenever a new window is created?

**Reply by LordDragonWood** [reply]
This one worked for me, thanks. I'm running WezTerm on PikaOS with KDE Plazma. So, maximize acts the same as fullscreen would on MacOS. Trying the fullscreen stuff didn't work for me.
